shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/resources/CustomResourceStatus.java

```diff
@@ -9,17 +9,16 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.redhat.service.smartevents.shard.operator.utils.DeploymentStatusUtils;
 
-import io.fabric8.kubernetes.api.model.apps.Deployment;
-import io.fabric8.kubernetes.client.internal.readiness.Readiness;
+import io.javaoperatorsdk.operator.api.ObservedGenerationAware;
 
 /**
  * Common interface for Kubernetes Custom Resource status
  */
-public abstract class CustomResourceStatus {
+public abstract class CustomResourceStatus implements ObservedGenerationAware {
```

Comment:
This is considered ["best practice"](https://javaoperatorsdk.io/docs/features#automatic-observed-generation-handling) and prevents an additional reconciliation loop from running now I've added an `ErrorStatusHandler`.

"Note that this update request will also produce an event, and will result in a reconciliation **if the controller is not generation aware**"